.github/workflows/test.yaml

```
@@ -66,21 +70,43 @@ jobs:
         env:
           FORCE_COLOR: 1
 
-      - name: Run tests with every array backend and generate coverage report
-        run: nox -s coverage-${{ matrix.python-version }} --verbose
+      - name: Run benchmark tests with all backends and generate coverage report
+        run: nox -s coverage_benchmarks-${{ matrix.python-version }} --verbose
         env:
+          ARRAY_BACKEND: all
+          COVERAGE_FILE: ${{ env.BENCH_COV_NAME }}-${{ matrix.python-version }}
           FORCE_COLOR: 1
+
+      - name: Run core tests with all backends and generate coverage report
+        run: nox -s coverage-${{ matrix.python-version }} --verbose
+        env:
           ARRAY_BACKEND: all
+          COVERAGE_FILE: ${{ env.CORE_COV_NAME }}-${{ matrix.python-version }}
+          FORCE_COLOR: 1
 
-      - name: Coveralls requires XML report
-        run: coverage xml
+      - name: Generate XML report for both sets of tests
+        run: >-
+          coverage xml --data-file ${{ env.BENCH_COV_NAME }}-${{
+          matrix.python-version }} -o ${{ env.BENCH_COV_NAME }}-${{
+          matrix.python-version }}.xml; coverage xml --data-file ${{
+          env.CORE_COV_NAME }}-${{ matrix.python-version }} -o ${{
+          env.CORE_COV_NAME }}-${{ matrix.python-version }}.xml;
```

I found this hard to read

```suggestion
      - name: Generate XML report for both sets of tests
        run: >-
          coverage xml --data-file ${{ env.BENCH_COV_NAME }}-${{
          matrix.python-version }} -o ${{ env.BENCH_COV_NAME }}-${{
          matrix.python-version }}.xml;
          coverage xml --data-file ${{ env.CORE_COV_NAME }}-${{
          matrix.python-version }} -o ${{ env.CORE_COV_NAME }}-${{
          matrix.python-version }}.xml;
```